cmd/dep/ensure.go

```diff
@@ -304,27 +304,22 @@ func getProjectConstraint(arg string, sm gps.SourceManager) (gps.ProjectConstrai
 }
 
 func checkErrors(m map[string]pkgtree.PackageOrErr) error {
-	noGoErrors, pkgErrors := 0, 0
-	for _, poe := range m {
+	buildErrors, noGoErrors := []string{}, []string{}
```

Comment:
I don't mind making the change, but these one-time allocations seem negligible in comparison to the rest of dep ensure's gc pressure.